RDKShell/RDKShell.cpp

```diff
@@ -198,6 +199,17 @@ namespace WPEFramework {
                         RdkShell::CompositorController::kill(service->Callsign());
                         RdkShell::CompositorController::removeListener(clientidentifier, mShell.mEventListener);
                         gRdkShellMutex.unlock();
+                        std::map<std::string, PluginStateChangeData*>::iterator pluginStateChangeEntry = gPluginsEventListener.find(service->Callsign());
+                        if (pluginStateChangeEntry != gPluginsEventListener.end())
+                        {
+                            PluginStateChangeData* stateChangeData = pluginStateChangeEntry->second;
+                            if (nullptr != stateChangeData)
+                            {
+                                delete stateChangeData;
+                            }
+                            pluginStateChangeEntry->second = nullptr;
+                            gPluginsEventListener.erase(pluginStateChangeEntry);
```

Comment:
Is there a risk of multiple threads accessing `gPluginsEventListener` at the same time?  Same concern about `gActivePlugins`.  Maybe it's best to protect with a new mutex to prevent a crash.